modules/afsocket/afsocket-dest.c

```diff
@@ -207,6 +233,46 @@ afsocket_dd_start_reconnect_timer(AFSocketDestDriver *self)
   iv_timer_register(&self->reconnect_timer);
 }
 
+// Returns TRUE in case of success, or there is nothing to do. FALSE will indicate an error.
+static gboolean
+_afsocket_dd_rename_connection_persist_name(AFSocketDestDriver *self, PersistState *state)
+{
+  const gchar *old_name, *new_name;
+  PersistEntryHandle old_handle, new_handle;
+  gsize size;
+  guint8 old_version, new_version;
+  gpointer old_value, new_value;
+
+  // Make sure there is no entry with the new name
+  new_name = afsocket_dd_format_connections_name(self);
+  new_handle = persist_state_lookup_entry(state, new_name, &size, &new_version);
+  if (new_handle)
+    return TRUE;
+
+  // Find the old entry
+  old_name = afsocket_dd_format_legacy_connection_name(self);
+  old_handle = persist_state_lookup_entry(state, old_name, &size, &old_version);
+  if (!old_handle)
+    return TRUE;
+
+  // Alloc a new entry. _restart_with_state will handle the "persistless" case.
+  new_handle = persist_state_alloc_entry(state, new_name, size);
+  if (!new_handle)
+    return FALSE;
+
+  old_value = persist_state_map_entry(state, old_handle);
+  new_value = persist_state_map_entry(state, new_handle);
+  memcpy(new_value, old_value, size);
+  persist_state_unmap_entry(state, old_handle);
+  persist_state_unmap_entry(state, new_handle);
+
+  msg_debug("Connections persist name changed",
+            evt_tag_str("old name", old_name),
+            evt_tag_str("new name", new_name));
+
```

Comment:
Shouldn't we remove the old entry?

Reply:
Not needed. The in_use bit logic will handle the old entry: https://github.com/balabit/syslog-ng/blob/bd6fac1eb892275e81562d93a33bdeb26ee5e833/lib/persist-state.c#L91

Actually the https://github.com/balabit/syslog-ng/blob/bd6fac1eb892275e81562d93a33bdeb26ee5e833/lib/persist-state.c#L760 function is just set this flag, and do not actually remove the entry.